### ISSUE TITLE
Progression counter ?

### ISSUE THREAD

**Issue by satsukiyatoshi** [wqweto/VszLib#2]
First, thanks for this great vb code.
I have a suggestion, when i need to deal with big archive (some Go) i have no idea of the unzip progression, plus the vb6 software freeze until the complete extraction.

>So any way to add some progression counter of the extraction/compression ?

Thanks

**Comment by wqweto** [comment]
You can take a look at the `Progress` event and provide UI with current filename + percent complete.

**Comment by satsukiyatoshi** [comment]
Thanks ! have you any sample code, because when i do a big file extraction the vb6 exe doesn't seems to respond anymore until end of extraction.

**Comment by wqweto** [comment]
Yes, you would need to call `DoEvents` or spin the message pump any other way manually, so that first, UI refreshes, and second, user is able to press cancel button or similar to interrupt extraction. The UI intricacies are not taken care by the library at all.

Currently I don't have any complete sample handy, you'll need to read the source and figure it out somehow. For instance on `FileIdx < 0` it seems the `Progress` event is reporting total extraction stats in `Current` and `Total`.
